app/src/org/commcare/activities/InstallArchiveActivity.java

```diff
@@ -149,11 +151,28 @@ private void onUnzipSuccessful() {
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
         if (requestCode == REQUEST_FILE_LOCATION && resultCode == Activity.RESULT_OK) {
-            // Android versions 4.4 and up sometimes don't return absolute
-            // filepaths from the file chooser. So resolve the URI into a
-            // valid file path.
-            String filePath = UriToFilePath.getPathFromUri(CommCareApplication.instance(),
-                    intent.getData());
+            updateFileLocationFromIntent(this, intent, editFileLocation);
+        }
+    }
+
+    public static void updateFileLocationFromIntent(Context context, Intent intent, EditText editFileLocation) {
+        // Android versions 4.4 and up sometimes don't return absolute
+        // filepaths from the file chooser. So resolve the URI into a
+        // valid file path.
+        Uri uriPath = intent.getData();
+        if (uriPath == null) {
+            // issue getting the filepath uri from file browser callout
+            // result
+            Toast.makeText(context,
+                    Localization.get("mult.install.state.invalid.path"),
+                    Toast.LENGTH_SHORT).show();
+        } else {
+            String filePath;
+            try {
+                filePath = UriToFilePath.getPathFromUri(CommCareApplication.instance(), uriPath);
+            } catch (UriToFilePath.NoDataColumnForUriException e) {
+                filePath = intent.getData().toString();

```

Comment:
Saving the filePath as content uri itself in case of an Exception, which gets displayed in edittext for filepath and later gets processed as Content uri